### PR TITLE
Dry up prepare pdf

### PR DESCRIPTION
**Pull request by leohemsted**

### CODE REVIEW

File: app/celery/tasks.py [hunk 1]
```diff
@@ -226,6 +205,15 @@ def create_pdf_for_templated_letter(self: Task, encoded_letter_data):
     )
 
 
+def _prepare_pdf(letter_details, self):
+    def create_pdf_for_letter(letter_details, language, include_tag) -> BytesIO:
```
Comment (CrystalPea) [inline app/celery/tasks.py:209]: Now I remembered - didn't we want to change this nested function and the same in `preview.py` into lambdas?

File: app/preview.py [hunk 2]
```diff
@@ -225,7 +217,7 @@ def get_html(json, language="english"):
 
 
 @sentry_sdk.trace
```
Comment (CrystalPea) [inline app/preview.py:219]: I wonder - do we still use this sentry tracing? Doesn't have to be removed in this PR, but might be a good time to think if we still need it?

File: app/celery/tasks.py (hunk 1 above)
Comment (leohemsted) [reply app/celery/tasks.py:209]: no, ruff will fail checks against lambdas assigned to variables, saying nested functions are preferred

we'd previously started with lambdas, then switched to get tests to pass, so i squashed that out.

looking elsewhere, there's some discussion saying "If you need to assign the lambda to a name, use a def instead"

https://stackoverflow.com/a/134638


i guess we could inline it, for example:

```
def _prepare_pdf(letter_details, self):
    purpose = PDFPurpose.PRINT

    return generate_templated_pdf(
        letter_details,
        (
            lambda letter_details, language, include_tag: _create_pdf_for_letter(
                self, letter_details, language=language, include_notify_tag=include_tag
            )
        ),
        purpose,
    )
```

but i don't particularly love that. open to hear other thoughts tho!

File: app/preview.py (hunk 2 above)
Comment (leohemsted) [reply app/preview.py:219]: hmmm i think lets keep this in here - i think i'd like a wider conversation about sentry before we decide what to do with these 🤔

File: app/celery/tasks.py (hunk 1 above)
Comment (CrystalPea) [reply app/celery/tasks.py:209]: Ok, no worries then ☺️

File: app/preview.py (hunk 2 above)
Comment (CrystalPea) [reply app/preview.py:219]: Okee 👍🏼

**Review by CrystalPea — APPROVED** [review]

**Pull request merged**